libraries/Microsoft.Bot.Configuration/BotConfigurationExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    public static class BotConfigurationExtensions
+    {
+        /// <summary>
+        /// Returns a IEnumerable of a specific <see cref="ConnectedService"/> type contained in the bot configuration file.
+        /// </summary>
+        /// <param name="botServices">BotConfiguration's services.</param>
+        /// <param name="serviceType">The type of <see cref="ConnectedService"/> to search.</param>
+        /// <returns>List of bot's <see cref="ConnectedService"/>.</returns>
+        public static IEnumerable<ConnectedService> OfType(this List<ConnectedService> botServices, string serviceType)
+        {
+            // If botServices is null, throws an exception
+            if (botServices == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            // If there aren't any service, returns an empty IEnumerable
+            if (botServices.Count == 0)
+            {
+                return Enumerable.Empty<ConnectedService>();
+            }
+
+            // Obtains all the services of the specified type
+            var botFilteredServices = botServices.Where(service => service.Type.Equals(serviceType, StringComparison.CurrentCultureIgnoreCase));
+
+            // If there isn't any service of the specified type, returns an empty IEnumerable
+            if (botFilteredServices == null || botFilteredServices.Count() == 0)
+            {
+                return Enumerable.Empty<ConnectedService>();
+            }
+
+            return botFilteredServices;
+        }
+
+        /// <summary>
+        /// Check if the Project's Name contains white spaces.
+        /// </summary>
+        /// <param name="botConfiguration">BotConfiguration object to extend.</param>
+        /// <param name="folder">Folder's path to validate.</param>
+        /// <exception cref="System.Exception"> Thrown when no '.scproj' file is found or when there are two or more 'csproj' files.</exception>
+        /// <exception cref="System.ArgumentNullException"> Thrown when param 'folder' is null.</exception>
+        /// <returns>Boolean that indicates if a Project's Name is spaceless.</returns>
+        public static bool IsProjectNameSpaceless(this BotConfiguration botConfiguration, string folder)
+        {
+            // If the parameter 'folder' is null or empty, throws an Exception
+            if (string.IsNullOrEmpty(folder))
+            {
+                throw new ArgumentNullException(nameof(folder));
+            }
+
+            var files = Directory.GetFiles(folder, "*.csproj", SearchOption.TopDirectoryOnly);
```

Comment:
Suggest using [`Directory::EnumerateFiles`](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.enumeratefiles?redirectedfrom=MSDN&view=netframework-4.7.2#System_IO_Directory_EnumerateFiles_System_String_System_String_System_IO_SearchOption_) instead of `GetFiles`. Then you would throw out the `Length` check and just do the `FirstOrDefault()` and check for `null` to throw the exception.